client/src/util/auth.jsx

```diff
@@ -30,243 +42,228 @@ function useAuthProvider() {
-          return new Promise(() => null);
-        })
-    );
+  const signin = async (email, password) => {
+    try {
+      const userCredential = await signInWithEmailAndPassword(
+        auth,
+        email,
+        password,
+      );
+
+      await handleAuth(userCredential.user);
+    } catch (error) {
+      handleError(error);
+    }
+    // return supabase.auth
+    //   .signInWithPassword({ email, password })
+    //   .then(handleError)
+    //   .then(handleAuth);
   };
 
-  const signinWithMagicLink = (email) => {
-    return supabase.auth
-      .signInWithOtp({
-        email,
-        options: {
-          emailRedirectTo: `${window.location.origin}/dashboard`,
-        },
-      })
-      .then(handleError);
+  const signinWithProvider = async (name) => {
+
+    let provider;
+
+    switch (name) {
+      case 'google':
+        provider = new GoogleAuthProvider();
+        break;
+      case 'github':
+        provider = new GithubAuthProvider();
+        break;
+      default:
+        throw new Error(`Provider ${name} is not supported`);
+    }
+
+    try {
+      await signInWithPopup(auth, provider);
+
+      window.location.href = `${window.location.origin}/dashboard`;
+    } catch (error) {
+      console.log('Error signing in with provider', error.message);
+      handleError(error);
+    }
+
+    return new Promise(() => null);
+
+    // return (
+    //   supabase.auth
+    //     .signInWithOAuth({
+    //       provider: name,
+    //       options: {
+    //         redirectTo: `${window.location.origin}/dashboard`,
+    //       },
+    //     })
+    //     .then(handleError)
+    //     // Because `signInWithOAuth` resolves immediately we need to add this so that
+    //     // it never resolves (component will display loading indicator indefinitely).
+    //     // Once social signin is completed the page will redirect to value of `redirectTo`.
+    //     .then(() => {
+    //       return new Promise(() => null);
+    //     })
+    // );
   };
 
-  const signout = () => {
-    return supabase.auth.signOut().then(handleError);
+  // const signinWithMagicLink = (email) => {
+  //   return supabase.auth
+  //     .signInWithOtp({
+  //       email,
+  //       options: {
+  //         emailRedirectTo: `${window.location.origin}/dashboard`,
+  //       },
+  //     })
+  //     .then(handleError);
+  // };
+
+  const signout = async () => {
+    signOut(auth)
+      .catch((error) => {
+        handleAuth(error);
+      });
+    // return supabase.auth.signOut().then(handleError);
   };
 
-  const sendPasswordResetEmail = (email) => {
-    return supabase.auth
-      .resetPasswordForEmail(email, {
-        redirectTo: `${window.location.origin}/auth/changepass`,
-      })
-      .then(handleError);
+  const sendPasswordResetEmailFunction = async (email) => {
+    return sendPasswordResetEmail(auth, email, {
+      url: `${window.location.origin}/auth/changepass`,
+    })
+      .then(handleError)
```

Comment:
If I am understanding your `handleError` correctly, if the `sendPasswordResetEmail()` returns any data, then this function always throws an error.

Remove the `.then(handleError)` since the `then` case shouldn't be an error case

Reply:
removed